src/main/java/io/cryostat/agent/WebServer.java

```diff
@@ -163,7 +163,8 @@ private HttpHandler wrap(HttpHandler handler) {
                 handler.handle(x);
             } catch (Exception e) {
                 log.error("Unhandled exception", e);
-                x.sendResponseHeaders(HttpStatus.SC_INTERNAL_SERVER_ERROR, 0);
+                x.sendResponseHeaders(
+                        HttpStatus.SC_INTERNAL_SERVER_ERROR, RemoteContext.BODY_LENGTH_NONE);
```

Comment:
should be RemoteContext.BODY_LENGTH_UNKNOWN

Reply:
Hmm. Should it? If there is an unhandled internal exception then I think what should happen is the Agent responds to the requesting client with an HTTP 500 (that's the `SC_INTERNAL_SERVER_ERROR`), and this response would only include headers. The response body is empty since there is no content to report when an exception occurred.

I guess it's possible that the endpoint context handler has already started writing a response to the context stream, then hits an exception and here we try to send a "no body" header when the response is already partially written. I'm not sure exactly how the agent's HTTP server handles that, but the requesting client should hopefully take that as a sign to drop the stream if it starts receiving any response body.

Reply:
ah I see - I thought you were just exchanging the numerical body lengths with constants

Reply:
Mostly - I think these two cases were actually bugs previously (probably because the 0/-1 constants are not very readable), and the new constant represents the correct behaviour.

Reply:
They are both references to the same constant, which form is used just depends on the lexical scope of the reference. Classes implementing the `RemoteContext` interface have the constant available in scope so they can use the bare `BODY_LENGTH_UNKNOWN`, whereas other classes like `WebServer` need to either `import i.c.a.r.RemoteContext` and use `RemoteContext.BODY_LENGTH_UNKNOWN`, or could `import static i.c.a.r.RemoteContext.BODY_LENGTH_UNKNOWN` and then use the bare reference as well.

Reply:
Hmm what about MBeanContext that implements RemoteContext but uses RemoteContext.BODY_LENGTH_UNKNOWN ?

Reply:
I can change that if you'd like, but it doesn't actually mean anything different.

Reply:
I just figured for consistency sake it might be less confusing, but otherwise everything looks good